.circleci/config.yml

```diff
@@ -4,37 +4,18 @@ workflows:
   workflow:
     jobs:
       - go-test:
```

Comment:
We had test jobs here for some really old Go versions, using obsolete CircleCI images. I've removed all the versions below 1.14 and added newer versions.